src/main/java/uk/gov/hmcts/darts/arm/api/impl/ArmDataManagementApiImpl.java

```diff
@@ -52,6 +53,11 @@ public UpdateMetadataResponse updateMetadata(String externalRecordId, OffsetDate
         return armApiService.updateMetadata(externalRecordId, eventTimestamp);
     }
 
+    @Override
+    public InputStream downloadArmData(String externalRecordId, String externalFileId) {
```

Comment:
Maybe good to pass a function into this method. That way the code can pass you the stream when it is available. That would naturally support both async or sync implementations as well as also transparently closing the IS for us. Just a thought

Reply:
this is a good idea Michael, I wanted to explore this solution. But addressing the other comments took more time than I thought so if you don't mind I'll leave this as is for now

Reply:
np, nice to have really